zqd/space/space.go

```diff
@@ -76,32 +76,53 @@ func (s Space) Name() string {
 }
 
 func (s Space) Info() (api.SpaceInfo, error) {
-	f, err := s.OpenFile(AllBzngFile)
+	logsize, err := s.LogSize()
 	if err != nil {
 		return api.SpaceInfo{}, err
 	}
-	defer f.Close()
-	stat, err := f.Stat()
+	packetsize, err := s.PacketSize()
 	if err != nil {
 		return api.SpaceInfo{}, err
 	}
 	spaceInfo := api.SpaceInfo{
-		Name:       s.Name(),
-		Size:       stat.Size(),
-		PacketPath: s.PacketPath(),
+		Name:          s.Name(),
+		Size:          logsize,
+		PacketSupport: s.PacketPath() != "",
+		PacketPath:    s.PacketPath(),
+		PacketSize:    packetsize,
 	}
-	i, err := loadInfo(s.conf.DataPath)
-	if err != nil {
-		if !os.IsNotExist(err) {
-			return api.SpaceInfo{}, err
-		}
-		return spaceInfo, nil
+	i, err := loadInfoFile(s.conf.DataPath)
+	if err == nil {
+		spaceInfo.MinTime = &i.MinTime
+		spaceInfo.MaxTime = &i.MaxTime
+	} else if !errors.Is(err, os.ErrNotExist) {
+		return api.SpaceInfo{}, err
 	}
+	return spaceInfo, nil
+}
 
-	spaceInfo.MinTime = &i.MinTime
-	spaceInfo.MaxTime = &i.MaxTime
+// LogSize returns the size in bytes of the logs in space.
+func (s Space) LogSize() (int64, error) {
+	return sizeof(s.DataPath(AllBzngFile))
+}
 
-	return spaceInfo, nil
+// PacketSize returns the size in bytes of the packet capture in the space.
+func (s Space) PacketSize() (int64, error) {
+	path := s.PacketPath()
+	if path == "" {
+		return 0, nil
+	}
+	return sizeof(path)
```

Comment:
If you want to, you can remove this check for "" since you check for os.ErrNotExist is sizeof.